### PR TITLE
💵 feat: claim (#102)

### LINKED ISSUE #102: Implement claim

### PR DESCRIPTION
**Pull request by aahna-ashina**
## Related GitHub Issue

close #102

## How Has This Been Tested?

- [x] Status checks pass (lint, build, test)
- [x] Works on `localhost`
- [ ] ~~Works on Goerli~~
- [x] Works on Sepolia - https://sepolia.etherscan.io/address/0xE0bE5f2282FC8A32758f914EDB3ED2727bE570E3
- [ ] Works on Mainnet

### CODE REVIEW

File: hardhat/contracts/BasicIncomeDistributor.sol [hunk 1]
```diff
@@ -157,9 +158,65 @@ contract BasicIncomeDistributor {
...
+        console.log("claimableAmount:", claimableAmount);
+        //TODO: Mark as claimed, then transfer
+        enrollmentTimestamps[msg.sender] = block.timestamp;
+        emit RewardClaimed(msg.sender, claimableAmount);
+        //ERC20(address(this)).transfer(msg.sender, claimableAmount);
+    }
+
+    function _getClaimableAmount(
+        address citizen
+    ) internal view returns (uint256) {
```
[the remark this answers is not in the document]
Comment (TTNguyenDev) [reply hardhat/contracts/BasicIncomeDistributor.sol:210]: Yes we only need one. I will update it

**Comment by TTNguyenDev** [comment]
@aahna-ashina , please review it. I'm working on new unit tests, but I think you can review the code first

File: hardhat/contracts/BasicIncomeDistributor.sol [hunk 2]
```diff
@@ -43,17 +44,22 @@ contract BasicIncomeDistributor {
...
+    IERC20 public token;
+
     error NotEligibleError(address citizen);
     error CurrentlyEnrolledError(address citizen, uint256 enrollmentTimestamp);
     error NotEnoughFunding(uint256 amountAvailable, uint256 amountRequested);
 
     constructor(
         address passportUtilsAddress,
         address nationCredAddress,
+        address rewardToken,
```
[the remark this answers is not in the document]
Comment (TTNguyenDev) [reply hardhat/contracts/BasicIncomeDistributor.sol:62]: Okay, understood, let me update it

[the remark this answers is not in the document]
Comment (aahna-ashina) [reply hardhat/contracts/BasicIncomeDistributor.sol]: Deleted here:  https://github.com/nation3/n3bi/pull/123/commits/88653939f6d4abb954481ba3677d44dbb72923ee

**Comment by TTNguyenDev** [comment]
Hi, I pushed my new code,  please take a look. Rewarded in ETH @aahna-ashina

**Pull request merged**